apps/sports-helsinki/src/domain/event/eventInfo/OrganizationInfo.tsx

```diff
@@ -21,12 +26,26 @@ interface Props {
 const OrganizationInfo: React.FC<Props> = ({ event }) => {
   const { t } = useTranslation('event');
   const locale = useLocale();
+  const router = useRouter();
   const { provider, publisher } = getEventFields(event, locale);
   const { data: organizationData, loading } = useOrganizationDetailsQuery({
     ssr: false,
     variables: { id: publisher },
   });
-
+  const searchParams = React.useMemo(
+    () => new URLSearchParams(qs.stringify(router.query)),
+    [router.query]
+  );
+  const returnPath = searchParams.get('returnPath') ?? '';
+  const returnPathSearchParams = new URLSearchParams(returnPath);
+  const returnPathSearchType = returnPathSearchParams.get('searchType') ?? '';
+  const eventTypeId: EventTypeId = isEventTypeId(returnPathSearchType)
+    ? returnPathSearchType
+    : EventTypeId.General; // Fallback value
+  const linkByPublisherLabel = React.useMemo(
```

Comment:
Just thinking, would `useCallback` do the same thing here and/or is somekind of caching/state needed? This is more like a question I'm just wondering if this and L35 would work the same as just normal vars..?

Reply:
Had some discussion about the differences of [useCallback ](https://reactjs.org/docs/hooks-reference.html#usecallback)and [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) and learned at least two things:
> useCallback(fn, deps) is equivalent to useMemo(() => fn, deps).

and

> You may rely on useMemo as a performance optimization, not as a semantic guarantee. In the future, React may choose to “forget” some previously memoized values and recalculate them on next render, e.g. to free memory for offscreen components. Write your code so that it still works without useMemo — and then add it to optimize performance.

Also discussed a bit about possible other ways to do this, maybe using state and useEffect etc. I think the conclusion was that although there are alternative ways to doing this, this is ok as it is now.